File: ios/Sources/RattleKit/RattlePreset.swift
import Foundation

public struct RattlePreset: Identifiable, Hashable, Sendable, Codable {
    public var id: String { name }

    public let category: RattleCategory
    public let name: String
    public let width: Int
    public let intervalMilliseconds: Int
    public let frames: [String]

    public init(
        category: RattleCategory,
        name: String,
        width: Int,
        intervalMilliseconds: Int,
        frames: [String]
    ) {
        self.category = category
        self.name = name
        self.width = width
        self.intervalMilliseconds = intervalMilliseconds
        self.frames = frames
    }

    public var interval: Duration {
        .milliseconds(intervalMilliseconds)
    }

    public var maxFrameLength: Int {
        frames.map(\.count).max() ?? 0
    }
}

File: android/src/main/java/io/rattlekit/RattlePreset.kt
package io.rattlekit

data class RattlePreset(
    val category: RattleCategory,
    val name: String,
    val width: Int,
    val intervalMillis: Int,
    val frames: List<String>,
) {
    val maxFrameLength: Int = frames.maxOfOrNull { it.length } ?: 0
}
